src/components/lib/cos_kernel_api.c

```diff
@@ -615,10 +715,29 @@ cos_mem_remove(pgtblcap_t pt, vaddr_t addr)
 	return 0;
 }
 
+vaddr_t
+cos_mem_move(struct cos_compinfo *dstci, struct cos_compinfo *srcci, vaddr_t src)
+{
+	vaddr_t dst;
+
+	assert(srcci && dstci);
+
+	dst = __page_bump_valloc(dstci);
+	if (unlikely(!dst)) return 0;
+
+	if (call_cap_op(srcci->pgtbl_cap, CAPTBL_OP_MEMMOVE, src, dstci->pgtbl_cap, dst, 0))  BUG();
+
+	return dst;
+}
+
 int
-cos_mem_move(pgtblcap_t ptdst, vaddr_t dst, pgtblcap_t ptsrc, vaddr_t src)
+cos_mem_move_at(struct cos_compinfo *dstci, vaddr_t dst, struct cos_compinfo *srcci, vaddr_t src)
```

Comment:
It would save the user some effort, and would mimic the `memcpy` API that people are comfortable with.

The downside is what to do when we get an error from the kernel after moving half of the entries.  But we would likely rather handle that in the library, than rely on the user of the library to do it likely incorrectly.